AirCasting/CoreData/PersistenceController.swift

```diff
@@ -6,11 +6,28 @@
 //
 
 import CoreData
+import SwiftUI
 
 class PersistenceController: ObservableObject {
     static let shared = PersistenceController()
 
+    var uiSuspended: Bool = false {
+        didSet {
+            Log.info("UI updates \(uiSuspended ? "suspended" : "resumed")")
+            if !uiSuspended { propagateChangesToUI() }
+        }
+    }
+    
+    private(set) lazy var viewContext: NSManagedObjectContext = {
+        let ctx = NSManagedObjectContext(concurrencyType: .mainQueueConcurrencyType)
+        ctx.parent = mainContext
+        ctx.automaticallyMergesChangesFromParent = true
+        return ctx
+    }()
+    
     private let container: NSPersistentContainer
+    
+    private lazy var mainContext: NSManagedObjectContext = container.newBackgroundContext()
```

Comment:
This name is misleading, it's not a main context ;)

Reply:
well, it is :) it's not main-thread context, but it indeed is our main context - the one that holds all the data and from which we spawn children. But I get your point - it might be a bit confusing beacause of the threading nature of core data. Will change it to something better